browser/ui/views/tabs/brave_tab_container.cc

```diff
@@ -297,11 +297,29 @@ void BraveTabContainer::PaintBoundingBoxForTiles(
 void BraveTabContainer::PaintBoundingBoxForTile(
     gfx::Canvas& canvas,
     const SplitViewBrowserData::Tile& tile) {
+  if (!GetTabCount()) {
+    return;
+  }
+
+  // Note that GetTabAtModelIndex() and IsValidModelIndex() is actually using
+  // "ViewModel" index. See TabContainerImpl::GetTabAtModelIndex(), and
+  // implementations in compound_tab_container.cc implementation. Thus, we need
+  // to add pinned tab count. Also NumPinnedTabsInModel() returns 0 when
   auto* tab_strip_model = tab_slot_controller_->GetBrowser()->tab_strip_model();
-  auto tab1_index = tab_strip_model->GetIndexOfTab(tile.first);
-  auto tab2_index = tab_strip_model->GetIndexOfTab(tile.second);
-  CHECK(controller_->IsValidModelIndex(tab1_index)) << tab1_index;
-  CHECK(controller_->IsValidModelIndex(tab2_index)) << tab2_index;
+  const bool is_pinned_tab_container =
+      tabs_view_model_.view_at(0)->data().pinned;
+  const int offset =
+      is_pinned_tab_container ? 0 : tab_strip_model->IndexOfFirstNonPinnedTab();
+
+  auto tab1_index = tab_strip_model->GetIndexOfTab(tile.first) - offset;
+  auto tab2_index = tab_strip_model->GetIndexOfTab(tile.second) - offset;
+  if (!controller_->IsValidModelIndex(tab1_index) ||
+      !controller_->IsValidModelIndex(tab2_index)) {
+    // In case the tiled tab is not in this container, this can happen.
+    // For instance, this container is for pinned tabs but the tile is
+    // unpinned.
```

Comment:
In this comment, `tile` is some what confusing.
IIUC, `tile` is a set of tabs for split view. (Am I right?)
If it's right, there is one pinned tab and unpinned tab in curren tile in this case.
`tile is unpinned` makes me confusing.

Reply:
Yes, `tabs in the tile is unpinned` would be better. The reason why I named it `tile` is that we want to have room for having multiple tab tiling, like 3+ tabs in a view.